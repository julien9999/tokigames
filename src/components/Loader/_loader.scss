.loader {
  &-wrapper {
    height: 100vh;
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  &-inner {
    width: 100%;
    text-align: center;
  }
}
