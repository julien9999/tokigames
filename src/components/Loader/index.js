import React from 'react';
import PropTypes from 'prop-types';

export default function Loader({ message }) {
  return (
    <div className="loader-wrapper">
      <div className="loader-inner">
        <div className="spinner-border" role="status">
          <span className="sr-only" />
        </div>
        <p>{message}</p>
      </div>
    </div>
  );
}

Loader.propTypes = {
  i18n: PropTypes.object,
};