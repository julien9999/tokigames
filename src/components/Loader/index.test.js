import React from 'react';
import renderer from 'react-test-renderer';
import Loader from './index';


describe('<Loader>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Loader message='Loading' />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

