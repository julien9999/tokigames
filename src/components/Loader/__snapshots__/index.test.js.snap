// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader> renders correctly 1`] = `
<div
  className="loader-wrapper"
>
  <div
    className="loader-inner"
  >
    <div
      className="spinner-border"
      role="status"
    >
      <span
        className="sr-only"
      />
    </div>
    <p>
      Loading
    </p>
  </div>
</div>
`;
